lib/ansible/modules/network/nxos/nxos_switchport.py

```diff
@@ -417,9 +388,13 @@ def vlan_range_to_list(vlans):
 def get_list_of_vlans(module):
 
     command = 'show vlan'
-    body = execute_show_command(command, module)
     vlan_list = []
-    vlan_table = body[0].get('TABLE_vlanbrief')['ROW_vlanbrief']
+
+    try:
+        body = execute_show_command(command, module)[0]
+        vlan_table = body['TABLE_vlanbrief']['ROW_vlanbrief']
+    except (KeyError, AttributeError, IndexError):
+        return None
```

Comment:
You later test `if vlan not in current_vlans`, so this should probably return `[]` (or something with defined membership) not `None`.